runtime/context.go

```diff
@@ -271,3 +296,97 @@ func (c *contextLogger) Warn(ctx context.Context, msg string, userFields ...zap.
 func (c *contextLogger) Check(lvl zapcore.Level, msg string) *zapcore.CheckedEntry {
 	return c.log.Check(lvl, msg)
 }
+
+// ContextMetrics emits metrics.
+type ContextMetrics interface {
+	GetOrAddInboundHTTPMetrics(ctx context.Context) *InboundHTTPMetrics
+	GetOrAddEndpointMetrics(ctx context.Context) *EndpointMetrics
+	GetOrAddInboundTChannelMetrics(ctx context.Context) *InboundTChannelMetrics
+	getEndpointMetrics(key string) *EndpointMetrics
```

Comment:
Are these method just for testing? If so, let's put some comments  and separate them from public methods.

Reply:
fixed